add private locations changelog (#120)
checkly/docs: changelog/private-locations.mdx

---
title: Private Locations Agent Changelog
sidebarTitle: Private Locations Agent
description: Release history for the Checkly Private Locations Agent
---

> Release history for the Checkly Private Locations Agent. For setup instructions, see the [Private Locations documentation](/platform/private-locations/overview).

Reviewer: Nit: this is repeated, so it reads a bit weirdly. I would remove the second occurrence of "Release history for the Checkly Private Locations Agent"

<img width="1087" height="488" alt="CleanShot 2025-12-15 at 18 13 49" src="https://github.com/user-attachments/assets/2c9e4f20-8ad8-495d-9673-e4c55ab4b581" />

> - **Major releases**: View summarized releases in the [product changelog](https://feedback.checklyhq.com/changelog?c=Checkly+Agent)

Reviewer: Nit: this looks a bit unpolished as a bullet point. Can we turn it into a callout?

```
<Note>
**Major releases**: View summarized releases in the [product changelog](https://feedback.checklyhq.com/changelog?c=Checkly+Agent)
</Note>
```

**Improvements**

- Minor patches on reporting

### Version 6.1.3 (August 14, 2025)

**Security updates**

- Improved secret scrubbing with new Playwright fork version
- Additional commits from runners
- Full changelog [here](https://github.com/checkly/checkly-runners/releases/tag/v6.1.3)

### Version 6.1.2 (August 6, 2025)

**Features**

- Add support for CHECKLY_SECRET environment variables
  - When an environment variable prefixed with `CHECKLY_SECRET_` contains special characters (anything other than A-Za-z0-9-_.~), the system now:
    1. Detects special characters using the regex pattern `/[^A-Za-z0-9\-_.~]/`
    2. Creates an encoded version using `encodeURIComponent()` with additional `!` character encoding (%21)
    3. Adds it to the scrubbing cache to ensure both original and encoded values are properly scrubbed from traces
  - This ensures that secrets containing special characters like `@, %, !, +`, etc. are properly scrubbed even when they appear in encoded form within request payloads or URLs

### Version 6.1.1 (July 30, 2025)

**Bug fixes**

- Make proxied env accessible to dependency callers

### Version 6.1.0 (July 25, 2025)

**Features**

> - Add support for secret detection on runtime for browser and MS checks (behind the Feature Flag `RUNTIME_SECRET_DETECTION`)

Reviewer: ```suggestion
- Add support for secret detection on runtime for Browser and Multistep checks for certain accounts
```

We should probably outright mention feature flags -- it won't make sense for most users

### Version 6.0.7 (June 30, 2025)

**Bug fixes**

> - Use lower case envars for cURL

Author: ```suggestion
- Use lower case environment variables for cURL
```

- Pass both headers
- Allow proxy to forward requests to minio
- Use xit

**Features**

> - Create new hash

Reviewer: What does this mean? This might be too much in the implementation detail weeds for users to understand.

- Configure with `runAsGroup: 10001` in your security context

**Stability fixes**

- Improved error handling for missing `await` statements in Playwright tests
- Fixed JSON report generation issues with oversized or unparsable values

### Version 6.0.5 (June 23, 2025)

**Features**

- Added support for [Playwright check suites](/detect/synthetic-monitoring/playwright-checks/overview).
- Added debug logging with `DEBUG=checkly:*` environment variable
- Bundled Node.js for offline installations

**Security updates**

- Enhanced secret scrubbing in logs
- Patched all reported npm vulnerabilities

**Improvements**

- Added retry logic for artifact uploads
- Improved fault tolerance

### Version 6.0.0 (May 9, 2025)

**Breaking changes**

- Minimum Docker version 20.10 required
- Updated configuration schema

**New features**

- Support for runtimes [2024.09](/platform/runtimes/runtime-specification#202409) and [2025.04](/platform/runtimes/runtime-specification#202504)
- Disabled `PLAYWRIGHT_NO_COPY_PROMPT` for better reporter compatibility

## Version history

- **Summary**: [Product changelog](https://feedback.checklyhq.com/changelog?c=Checkly+Agent)

> - **Latest versions published**: [Docker Hub tags](https://hub.docker.com/r/checkly/agent/tags)

Reviewer: This section is pretty small, and we already link the Product Changelog at the beginning of the page. 

Would it be better to remove this section and also link the Docker Hub tags at the beginning of the page?

## Compatibility matrix

| Agent version | Docker minimum | Supported runtimes | Node.js |
|--------------|----------------|-------------------|---------|
| 6.x | 20.10 | [2025.04](/platform/runtimes/runtime-specification#202504), [2024.09](/platform/runtimes/runtime-specification#202409) | v22.11.0 |
| 5.x | 19.03 | [2025.04](/platform/runtimes/runtime-specification#202504) | v22.11.0 |
| 4.x | 19.03 | [2024.09](/platform/runtimes/runtime-specification#202409) | v18 |

> | 3.x | 19.03 | [2023.09](/platform/runtimes/runtime-specification#202309) | v18 |

Reviewer: I think we should remove the "Supported runtimes" column and link to [/platform/private-locations/agent-configuration#agent-version-and-runtimes](https://www.checklyhq.com/docs/platform/private-locations/agent-configuration/#agent-version-and-runtimes) instead (the other page is more precise)

Reviewer 2: Agreed, made this change in my last commit.

## Update checklist

When [updating to the latest agent](/platform/private-locations/agent-configuration), ensure:

1. Confirm Docker version compatibility
2. Review breaking changes like Node.js supported version.
3. Test in staging environment
4. Update configuration files
5. Deploy changes

## Getting help

- View logs with `docker logs <container-id>`

> - Enable debug mode for detailed diagnostics

Reviewer: What's this "debug mode" referring to?

Reviewer 2: IMO this section feels a bit redundant, I'd be in preference on keeping the change log concise. Removed it in my latest commit but happy to add it back in with the tweaks you suggested, if you prefer.

> - Join the global [developer community](https://www.checklyhq.com/slack/) discussions

Reviewer: I think we should be clearer about what this links to. I would this to say "Join the Checkly Slack community" or similar

> - [Contact the support team](https://app.checklyhq.com/?support=true) for help

Reviewer: Could we change this to the support@checklyhq.com email?

Users who aren't logged in will get sent to a login page, and I don't want anyone to think they need to log in for support

Reviewer 2: Same as below: IMO this section feels a bit redundant, I'd be in preference on keeping the change log concise. Removed it in my latest commit but happy to add it back in with the tweaks you suggested, if you prefer.

---

Release history for the Checkly Private Locations Agent. For setup instructions, see the [Private Locations documentation](/platform/private-locations/overview).

- **Major releases**: View summarized releases in the [product changelog](https://feedback.checklyhq.com/changelog?c=Checkly+Agent)

## Checkly Agent Releases

### Version 6.1.7 (December 3, 2025)

**Features**

- First image also including a developer version `6.1.7-dev`, learn all about the [dev image](https://www.checklyhq.com/docs/platform/private-locations/dev-agent/) here.
  - From now on, you can get the most recent dev image available using `checkly/agent:dev`.
- Playwright Check Suites: improve internal logging and metrics.
- Browser and MultiStep checks using Runtime 2025-04 automatically scrub the authorization header in requests.

### Version 6.1.6 (November 24, 2025)

**Features**

- Added Checkly environment variables and configuration to [Playwright Check Suites](/detect/synthetic-monitoring/playwright-checks/overview). See the docs on [Playwright Check Suites environment variables](/detect/synthetic-monitoring/playwright-checks/environment-variables)

>   - Add support for `CHECKLY` to all checkly triggered runs: scheduled checks and test sessions triggered through either `npx checkly test`, `npx checkly trigger` or `npx checkly pw-test`

Author: adds given the tense we're agreeing on :)

>   - Include the environment variable `CI` as true only for test session triggered runs

Reviewer 3: ```suggestion
  - Included the environment variable `CI` as true only for test session triggered runs
```

Reviewer 3: Or "Includes"?

### Version 6.1.5 (November 14, 2025)

**Features**

- Added playwright test metrics support for Test performance and Test statuses graphs in Check Overview

### Version 6.1.4 (October 29, 2025)

**Improvements**

- Minor patches on reporting

### Version 6.1.3 (August 14, 2025)

**Security updates**

- Improved secret scrubbing with new PW fork version
- Additional commits from runners
- Full changelog [here](https://github.com/checkly/checkly-runners/releases/tag/v6.1.3)

### Version 6.1.2 (August 6, 2025)

**Features**

- Add support for CHECKLY_SECRET environment variables
  - When a CHECKLY_SECRET_ environment variable contains special characters (anything other than A-Za-z0-9-_.~), the system now:

>     1. Detects special characters using regex pattern `/[^A-Za-z0-9\-_.~]/`

Reviewer 3: ```suggestion
    1. Detects special characters using the regex pattern `/[^A-Za-z0-9\-_.~]/`
```
